File: src/agent/python/options.py
import os
import platform
from typing import cast

# Paths options
if platform.system() == "Windows":
    platform_specific_root = os.path.join(cast(str, os.getenv('LOCALAPPDATA')), "nexus-agent")

else:
    platform_specific_root = os.path.join(cast(str, os.getenv("HOME")), ".local", "share", "nexus-agent")

config_folder_path = os.getenv("NEXUSAGENT_PATHS__CONFIG", default=os.path.join(platform_specific_root, "config"))
packages_folder_path = os.getenv("NEXUSAGENT_PATHS__PACKAGES", default=os.path.join(platform_specific_root, "packages"))

json_rpc_listen_address = os.getenv("NEXUSAGENT_SYSTEM__JSONRPCLISTENADDRESS", default="0.0.0.0")
json_rpc_listen_port = int(os.getenv("NEXUSAGENT_SYSTEM__JSONRPCLISTENPORT", default="56145"))

File: src/agent/dotnet/Core/Options.cs
using System.Runtime.InteropServices;

namespace Nexus.Agent.Core;

internal abstract record NexusAgentOptions()
{
    // for testing only
    public string? BlindSample { get; set; }

    internal static IConfiguration BuildConfiguration()
    {
        var environmentName = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");

        var builder = new ConfigurationBuilder()
            .AddJsonFile("appsettings.json");

        if (!string.IsNullOrWhiteSpace(environmentName))
        {
            builder
                .AddJsonFile($"appsettings.{environmentName}.json", optional: true, reloadOnChange: true);
        }

        builder.AddEnvironmentVariables(prefix: "NEXUSAGENT_");

        return builder.Build();
    }
}

internal record SystemOptions : NexusAgentOptions
{
    public const string Section = "System";

    public string JsonRpcListenAddress { get; set; } = "0.0.0.0";

    public int JsonRpcListenPort { get; set; } = 56145;
}

internal record PathsOptions : IPackageManagementPathsOptions
{
    public const string Section = "Paths";

    public string Config { get; set; } = Path.Combine(PlatformSpecificRoot, "config");

    public string Packages { get; set; } = Path.Combine(PlatformSpecificRoot, "packages");

    private static string PlatformSpecificRoot { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
        ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "nexus-agent")
        : Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".local", "share", "nexus-agent");
}